src/base/bittorrent/session.h

```diff
@@ -727,7 +729,7 @@ namespace BitTorrent
         QString m_resumeFolderPath;
         QFile *m_resumeFolderLock = nullptr;
 
-        QTimer *m_refreshTimer = nullptr;
+        bool m_sessionRefreshed = true;
```

Comment:
Maybe there could be a better name? session is definitely not refreshed at the time of class construction.

Reply:
Seems completely redundant for me. Why not just shedule refresh in Session constructor and then do it in alert handler?

Reply:
@glassez I assume you're saying to just schedule the update just after handling the individual alerts
but you see we wait for two alerts https://github.com/qbittorrent/qBittorrent/blob/2a638d16e023968ea501f491be508d442fdf723f/src/base/bittorrent/session.cpp#L4146

and if we were to separate them the time gap between them will just grow, this will just worsen the bugs like these https://github.com/qbittorrent/qBittorrent/issues/12459

Reply:
Ok, you're right.
It seems that we could still have a simpler logic if we are sure that alerts would always be received in the same order...
>if we were to separate them the time gap between them will just grow, this will just worsen the bugs like these #12459

It seems like the only way to really fix it is to count all the summarized values at qBittorrent side based on the individual values of each torrent (or ask @arvidn to add a "merged" code to the libtorrent so that we can query both torrents and session statistics simultaneously).

Reply:
>Maybe there could be a better name? session is definitely not refreshed at the time of class construction.

`m_refreshEnqueued`.
Initially `false`.
Then:
```c++
void Session::enqueueRefresh()
{
    Q_ASSERT(!m_refreshEnqueueded);

    QTimer::singleShot(timeout, this, [this] ()
    {
        m_nativeSession->post_torrent_updates();
        m_nativeSession->post_session_stats();
    });

    m_refreshEnqueueded = true;
}
```
in alert handlers:
```c++
if (m_refreshEnqueueded)
    m_refreshEnqueueded = false;
else
    enqueueRefresh();
```
(putting the `if-else` logic above in the `enqueueRefresh` method makes the code a little shorter (which may seem like a good idea), but it makes the general logic less readable, which can't be justified by shortening a couple of lines of code)